### PR TITLE
feature/custom visibility algorithm

### PR DESCRIPTION
**Pull request by theiskaa**
Resolves: #26

### CODE REVIEW

**Comment by theiskaa** [comment]
Here is the example how you can implement:

```dart
bottomNavigationBar: Hidable(
  controller: scrollController,
  enableOpacityAnimation: true, // As default it's true
  hidableVisibility: (position, previousOffset, currentOffset, currentVisibility) {
    // Calculate the visibility with position, previousOffset, currentOffset, and
    // currentVisibility.
    return 1; // Set to always show.
  },
  child: BottomNavigationBar(
    currentIndex: index,
    onTap: (i) => setState(() => index = i),
    items: bottomBarItems(),
  ),
),
```

File: lib/src/hidable_controller.dart [hunk 1]
```diff
@@ -70,56 +62,34 @@ typedef HidableVisibility = double Function(
...
   ScrollController scrollController;
-  double size;
   HidableVisibility? hideableVisibility;
+  double deltaFactor;
 
   HidableController({
     required this.scrollController,
-    required this.size,
     this.hideableVisibility,
+    this.deltaFactor = 0.08,
```
Comment (radibobovich) [inline lib/src/hidable_controller.dart:71]: I tinkered with the delta and 0.08 seemed like too fast for me, I set it to 0.04 after my issue message.

**Review by radibobovich — APPROVED** [review]
Looks fine, I tested and it works as expected

File: lib/src/hidable_controller.dart (hunk 1 above)
Comment (theiskaa) [reply lib/src/hidable_controller.dart:71]: Cool, let me test. If seems fine I'll make the default 0.04 also. But I think, the current one works fine regarding speed.

Comment (theiskaa) [reply lib/src/hidable_controller.dart:71]: @radeonovich gonna go with 0.06, 4 seemd very slow and 8 was fast right. I tested with the 0.06 and it seems ok.

**Pull request merged**